app/ts/components/pages/WebsiteAccess.tsx

```diff
@@ -0,0 +1,561 @@
+	)
+}
+
+type BigIntToEthereumAddress = Omit<JSX.HTMLAttributes<HTMLDataElement>, 'value'> & { bigIntAddress: bigint }
+const BigIntToEthereumAddress = ({ bigIntAddress, onClick, ...props }: BigIntToEthereumAddress) => {
+	const ethereumAddress = serialize(EthereumAddress, bigIntAddress)
+	return <data { ...props } value = { ethereumAddress }>{ ethereumAddress }</data>
+}
+
+const AdvancedSettings = ({ websiteAccess }: { websiteAccess: Signal<WebsiteAccess> }) => {
+	return (
+		<Collapsible summary = 'Advanced Settings' defaultOpen>
+			<BlockRequestSetting websiteAccess = { websiteAccess } />
+			<DisableProtectionSetting websiteAccess = { websiteAccess } />
+			<RemoveWebsiteSetting websiteAccess = { websiteAccess } />
+		</Collapsible>
+	)
+}
+
+
+const BlockRequestSetting = ({ websiteAccess }: { websiteAccess: Signal<WebsiteAccess> }) => {
+	const requestBlockMode = websiteAccess.value.declarativeNetRequestBlockMode
+
+	const setWebsiteExternalRequestBlocking = async (shouldBlock: boolean) => {
+		sendPopupMessageToBackgroundPage({ method: 'popup_blockOrAllowExternalRequests', data: { website: websiteAccess.value.website, shouldBlock } })
+	}
+
+	const confirmOrRejectRequestBlocking = (response: string) => {
+		if (response !== 'confirm') return
+		setWebsiteExternalRequestBlocking(true)
+	}
+
+	if (!websiteAccess.value.access) return <></>
+
+	return (
+		<article class = 'flexy flexy-lg'>
+			<figure><i class = 'status-lg status-warn'><RequestBlockedIcon /></i></figure>
+			<section class = 'flexy' style = { { flex: 1, '--pad-y': 0 } }>
+				<div style = { { contain: 'inline-size', flex: '1 20ch', marginBottom: '0.5rem' } }>
+					<h1 style = { { color: 'var(--text-color)', whiteSpace: 'nowrap' } }>Block External Request</h1>
+					<p style = { { color: 'var(--disabled-text-color)', fontSize: '0.875rem' } }>The Interceptor can block network requests from this domain, effectively preventing the website from dialing to unknown domains and services.</p>
+				</div>
+				<aside>
+
+					{ requestBlockMode === 'block-all' ? (
+						<button class = 'btn btn--primary' onClick = { () => setWebsiteExternalRequestBlocking(false) }><span style = { { whiteSpace: 'nowrap' } }>Unblock Requests</span></button>
+					) : (
+						<Modal>
+							<Modal.Open class = 'btn btn--destructive'><span style = { { whiteSpace: 'nowrap' } }>Block Requests</span></Modal.Open>
+							<Modal.Dialog class = 'dialog' style = { { textAlign: 'center', color: 'var(--disabled-text-color)' } } onClose = { confirmOrRejectRequestBlocking }>
+								<h2 style = { { fontWeight: 600, fontSize: '1.125rem', color: 'var(--text-color)', marginBlock: '1rem' } }>Confirm Blocking External Requests</h2>
+								<p></p>
+								<p style = { { marginBlock: '0.5rem' } }>This will prevent <pre>{ websiteAccess.value.website.websiteOrigin }</pre> from requesting resources outside its domain, which can lead to erratic behavior or even cause it to stop functioning entirely.</p>
```

Comment:
we should show the website name + icon for this. There's component for it

Reply:
website urls will have their icons on them throughout the page in #1117